traitlets/traitlets.py

```diff
@@ -908,6 +965,32 @@ def unobserve_all(self):
         """Remove all trait change handlers."""
         self._trait_notifiers = {}
 
+    def register_validator(self, handler, name):
```

Comment:
I thought we decided to only support decorators for HasTraits methods, which would mean that register_validator would be a private detail.  @ellisonbg, what do you think?